File: Test/AdventureWorksFunctionalModel/Mapping/ScrapReasonMap.cs
using System.Data.Entity.ModelConfiguration;
using AW.Types;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace AW.Mapping {
    public class ScrapReasonMap : EntityTypeConfiguration<ScrapReason> {
        public ScrapReasonMap() {
            // Primary Key
            HasKey(t => t.ScrapReasonID);

            // Properties
            Property(t => t.Name)
                .IsRequired()
                .HasMaxLength(50);

            // Table & Column Mappings
            ToTable("ScrapReason", "Production");
            Property(t => t.ScrapReasonID).HasColumnName("ScrapReasonID");
            Property(t => t.Name).HasColumnName("Name");
            Property(t => t.ModifiedDate).HasColumnName("ModifiedDate"); //.IsConcurrencyToken();
        }
    }

    public static partial class Mapper {
        public static void Map(this EntityTypeBuilder<ScrapReason> builder) {
            builder.HasKey(t => t.ScrapReasonID);

            // Properties
            builder.Property(t => t.Name)
                   .IsRequired()
                   .HasMaxLength(50);

            // Table & Column Mappings
            builder.ToTable("ScrapReason", "Production");
            builder.Property(t => t.ScrapReasonID).HasColumnName("ScrapReasonID");
            builder.Property(t => t.Name).HasColumnName("Name");
            builder.Property(t => t.ModifiedDate).HasColumnName("ModifiedDate"); //.IsConcurrencyToken();
        }
    }
}

File: Test/AdventureWorksLegacyModelVB/Mapping/ScrapReasonMap.vb
Imports System.Data.Entity.ModelConfiguration

Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Metadata.Builders

Namespace AW.Mapping
	Public Class ScrapReasonMap
		Inherits EntityTypeConfiguration(Of ScrapReason)

		Public Sub New()
			' Primary Key
			HasKey(Function(t) t.ScrapReasonID)

			' Properties
			[Property](Function(t) t.Name).IsRequired().HasMaxLength(50)

			' Table & Column Mappings
			ToTable("ScrapReason", "Production")
			[Property](Function(t) t.ScrapReasonID).HasColumnName("ScrapReasonID")
			[Property](Function(t) t.Name).HasColumnName("Name")
			[Property](Function(t) t.ModifiedDate).HasColumnName("ModifiedDate") '.IsConcurrencyToken();
		End Sub
	End Class

	Public Module Mapper
		<System.Runtime.CompilerServices.Extension> _
		Public Sub Map(ByVal builder As EntityTypeBuilder(Of ScrapReason))
			builder.HasKey(Function(t) t.ScrapReasonID)

			' Properties
			builder.Property(Function(t) t.Name).IsRequired().HasMaxLength(50)

			' Table & Column Mappings
			builder.ToTable("ScrapReason", "Production")
			builder.Property(Function(t) t.ScrapReasonID).HasColumnName("ScrapReasonID")
			builder.Property(Function(t) t.Name).HasColumnName("Name")
			builder.Property(Function(t) t.ModifiedDate).HasColumnName("ModifiedDate") '.IsConcurrencyToken();
		End Sub
	End Module
End Namespace